dashboard/test/ui/features/step_definitions/gamelab.rb

```diff
@@ -57,11 +57,12 @@
 end
 
 Then /^I select the animal category of the animation library$/ do
-  @browser.execute_script("$(\"img[src*='/category_animals.png']\")[0].click();")
+  @browser.execute_script("$(\"img[src*='/category_animals.png']\")[1].click();")
```

Comment:
if we wait until we see the relevant element (same as you did below), can we remove the new `And I wait for 2 seconds` step from the UI test? it's probably unlikely, but the fixed wait time could introduce flakiness

Reply:
Is there a way in the UI test to wait for a network response? The button to open the modal is already there, but we don't want to open it until the manifest is loaded (this isn't an issue for users because it loads fast, just not as fast as the UI test)

Reply:
i'm confused -- doesn't the presence of the `category_animals.png` image mean that the manifest has loaded?

Reply:
`P5LabView` loads the manifest in `componentDidMount`, then passes a getter to `AnimationPicker`. But that means that if you open the animation picker before the manifest loads, it won't show any of the categories (and they don't pop in after, I think because React doesn't think it needs to re-render `AnimationPicker` since none of its props/state changed). So we need to make sure we don't open the modal until after the manifest has loaded

Reply:
ohhhh, i see now. there may be a way to wait for a specific network response via selenium, but i'm not sure. i see a couple of examples of us making browser requests, but i can't find any that are waiting for in-flight requests:
https://github.com/code-dot-org/code-dot-org/blob/55a9a44703dba25d041c9b2cd8e5258e759c536e/dashboard/test/ui/features/step_definitions/steps.rb#L972

ideally, i think we would refactor the `getManifest` function in `<P5LabView/>` so that it returns a promise and either loads the manifest or returns a cached manifest, which would fix this really unlikely race condition. that's somewhat unrelated to what you're doing here though, so i think it's your call

Reply:
It looks like you can use `wait_until` to wait until a JS variable is a certain value, or a function returns a certain value, so if you can expose your manifest loaded state in one of these ways, it would become easy to delay.